### PR TITLE
feat: add Stepflow-Langflow integration

### PR DESCRIPTION
**Pull request by bjchambers**
This commit introduces a complete integration between Stepflow and Langflow, enabling seamless conversion and execution of visual AI workflows.

**Langflow Converter & Translator**:
- JSON workflow parsing and dependency analysis
- Component node processing with UDF executor generation
- Schema mapping and type conversion
- YAML workflow generation for Stepflow

**UDF Execution Engine**:
- Dynamic Langflow component execution through blob storage
- Type conversion between Langflow and Stepflow formats
- Bidirectional communication with Stepflow runtime
- Memory management and context preservation

**Component Server**:
- StepflowStdioServer integration for plugin communication
- JSON-RPC protocol support with proper type annotations
- Component registration and lifecycle management

**Testing Infrastructure**:
- Real Stepflow binary integration for validation
- Comprehensive test expectations framework
- Performance benchmarking for different workflow sizes
- Error handling and edge case validation

Successfully converts and executes **5 Langflow starter projects**:

- **Basic Prompting** (6 nodes): ChatInput → PromptTemplate → LanguageModel
- **Memory Chatbot** (6 nodes): Conversation memory and multi-turn chat
- **Document Q&A** (6 nodes): File processing and document analysis
- **Simple Agent** (7 nodes): Agent with Calculator and URL tools
- **Vector Store RAG** (17 nodes): Complex RAG pipeline with embeddings

**Component Support**: ChatInput, ChatOutput, PromptTemplate, LanguageModel, Agent, Calculator, URL, Memory, VectorStore, TextSplitter, Embeddings, File components

**UDF Execution Model**: Langflow components → UDF executors with blob storage
**Dependency Resolution**: Automatic step dependency generation from workflow edges
**Type System**: Full compatibility with Langflow Message/Data/DataFrame types
**Plugin Integration**: Native Stepflow plugin architecture support

- **CLI Interface**: `stepflow-langflow convert/analyze/validate` commands
- **Python API**: Programmatic conversion and analysis capabilities
- **Documentation**: Comprehensive examples and compatibility matrix
- **Testing Framework**: Real binary validation and performance benchmarks

This integration bridges the gap between visual AI workflow design (Langflow) and production execution (Stepflow), enabling seamless migration of existing Langflow workflows to the Stepflow ecosystem.

### CODE REVIEW

File: integrations/langflow/PLAN.md
Comment (bjchambers) [inline integrations/langflow/PLAN.md:1]: Remove PLAN.md

File: integrations/langflow/README.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+All standard Langflow components are supported, including:
+
+- **Chat Components**: ChatInput, ChatOutput, LanguageModel
+- **Data Processing**: TextSplitter, DocumentLoader, VectorStore
+- **AI Models**: OpenAI, Anthropic, Google models
+- **Custom Components**: User-defined components via UDF execution
+
+## Development
+
+See [PLAN.md](./PLAN.md) for detailed implementation plan and development guide.
```
Comment (bjchambers) [inline integrations/langflow/README.md:96]: Update.

File: integrations/langflow/examples/basic_conversion.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python3
+"""Basic conversion example for Langflow to Stepflow integration."""
+
+import json
+from pathlib import Path
+
+from stepflow_langflow_integration import LangflowConverter
+
+
+def main():
```
Comment (bjchambers) [inline integrations/langflow/examples/basic_conversion.py:10]: Does this example add value? Or should we just update the redame to talk about how to use the CLI?

File: integrations/langflow/examples/langflow_examples.md
Comment (bjchambers) [inline integrations/langflow/examples/langflow_examples.md:1]: If we get rid of langflow_examples should add some of the content to the README.md.

File: integrations/langflow/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    "stepflow-py>=0.2.4",
+    "pyyaml>=6.0",
+    "msgspec>=0.19.0",
+    "click>=8.0",
+    "jsonschema>=4.17.0",
+    "langflow>=1.5.0",  # Required for component execution
+]
+
+[project.optional-dependencies]
+# Future enhancement: Graph API fallback for parsing
```
Comment (bjchambers) [inline integrations/langflow/pyproject.toml:48]: Remove graph fallback.

File: integrations/langflow/src/stepflow_langflow_integration/cli/main.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        click.echo(f"❌ Conversion failed: {e}", err=True)
+        sys.exit(1)
+    except Exception as e:
+        click.echo(f"❌ Unexpected error: {e}", err=True)
+        sys.exit(1)
+
+
+@main.command()
+@click.argument("input_file", type=click.Path(exists=True, path_type=Path))
+def analyze(input_file: Path):
```
Comment (bjchambers) [inline integrations/langflow/src/stepflow_langflow_integration/cli/main.py:54]: Add explanation of what analyze does? It seems to attempt to predict errors?

File: integrations/langflow/src/stepflow_langflow_integration/converter/schema_mapper.py [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+            "dict": "object",
+            "dropdown": "string",
+            "slider": "number",
+            "file": "string",
+            "code": "string",
+            "prompt": "string",
+            "multiline": "string",
+        }
+        
+        self.component_output_heuristics = {
```
Comment (bjchambers) [inline integrations/langflow/src/stepflow_langflow_integration/converter/schema_mapper.py:26]: What do the component output heuristics do?

> Set the output types for components not covered by outputs or base classes.

File: integrations/langflow/tests/integration/test_langflow_integration.py [hunk 6]
```diff
@@ -0,0 +1,352 @@
...
+                )
+                
+            except (ConversionError, ValidationError):
+                # Skip performance test for workflows that don't convert
+                continue
+    
+    @pytest.mark.slow
+    @pytest.mark.asyncio
+    @patch('stepflow_langflow_integration.executor.udf_executor.UDFExecutor._execute_langflow_component')
+    async def test_execution_with_mocking(
```
Comment (bjchambers) [inline integrations/langflow/tests/integration/test_langflow_integration.py:231]: Todo what is this testing? It seems to be mocking the UDF executor at the component level? How does this actually inject these mock behaviors?

File: integrations/langflow/tests/integration/test_stepflow_binary_validation.py [hunk 7]
```diff
@@ -0,0 +1,453 @@
...
+        if not success:
+            # Langflow components might not be available
+            if "langflow" in stderr.lower() or "not found" in stderr.lower():
+                pytest.skip(f"Langflow components not available in test environment: {stderr}")
+            else:
+                pytest.fail(f"Workflow execution failed unexpectedly:\nSTDOUT: {stdout}\nSTDERR: {stderr}")
+        
+        # If execution succeeded, validate result structure
+        assert result_data is not None
+        assert isinstance(result_data, dict)
```
Comment (bjchambers) [inline integrations/langflow/tests/integration/test_stepflow_binary_validation.py:197]: Need tighter assertions on this -- this just verifies that the execution succeeds, not what is actually produced.

**Comment by zznate** [comment]
Also, sign the ICLA lol

File: integrations/langflow/src/stepflow_langflow_integration/cli/main.py [hunk 8]
```diff
@@ -0,0 +1,405 @@
...
+    except (ConversionError, ValidationError) as e:
+        click.echo(f"❌ Analysis failed: {e}", err=True)
+        sys.exit(1)
+    except Exception as e:
+        click.echo(f"❌ Unexpected error: {e}", err=True)
+        sys.exit(1)
+
+
+@main.command()
+@click.option("--host", default="localhost", help="Server host")
```
Comment (bjchambers) [inline integrations/langflow/src/stepflow_langflow_integration/cli/main.py:124]: Should add a more explicit option for "stdio" vs. "http". Note that we don't yet support http.

File: sdks/python/src/stepflow_py/flow_builder.py [hunk 9]
```diff
@@ -260,13 +265,199 @@ def set_output(self, output_data: Valuable) -> FlowBuilder:
...
+        
+        Args:
+            base_name: Base name for the step
+            
+        Returns:
+            Unique step ID, possibly with namespace prefix
+        """
+        self._step_counter += 1
+        if self.namespace:
+            return f"{self.namespace}_{base_name}_{self._step_counter}"
```
Comment (bjchambers) [inline sdks/python/src/stepflow_py/flow_builder.py:301]: Eliminate the namespace? Seems to add unnecessary complexity.

File: integrations/langflow/src/stepflow_langflow_integration/converter/dependency_analyzer.py [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+
+
+class DependencyAnalyzer:
+    """Analyzes Langflow edges to build step dependency graphs."""
+    
+    def build_dependency_graph(self, edges: List[Dict[str, Any]]) -> Dict[str, List[str]]:
+        """Build a dependency graph from Langflow edges.
+        
+        Args:
+            edges: List of Langflow edge objects
```
[the remark this answers is not in the document]
Comment (bjchambers) [reply integrations/langflow/src/stepflow_langflow_integration/converter/dependency_analyzer.py:28]: This is an implementation detail of the conversion. Not sure why we would link to the README or what we would link to?

File: integrations/langflow/test_comprehensive.py [hunk 11]
```diff
@@ -0,0 +1,309 @@
...
+from pathlib import Path
+from typing import Dict, List, Tuple, Optional
+import subprocess
+import tempfile
+import time
+
+# Add src to path for imports
+sys.path.insert(0, str(Path(__file__).parent / "src"))
+
+class ComprehensiveTestSuite:
```
Comment (bjchambers) [inline integrations/langflow/test_comprehensive.py:20]: Is this useful/necessary?

File: integrations/langflow/README.md [hunk 12]
```diff
@@ -0,0 +1,472 @@
...
+## 📚 Examples
+
+The `tests/fixtures/langflow/` directory contains example workflows from official Langflow starter projects. **All examples work successfully in `--mock` mode** (7/7) and **6 out of 7 workflows work in real execution mode** with API keys configured.
+
+### 🚀 Quick Test Commands
+
+All workflows accept a `"message"` field as their primary input. **Integration Status: 6/7 workflows fully working with real execution.**
+
+#### Mock Mode Testing (Safe - No API Keys Required)
+```bash
```
Comment (bjchambers) [inline integrations/langflow/README.md:309]: Update and/or remove?
